server/models/utils/sortContests.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @param {Array} param0.contests
+ * @param {Object} param0.query
+ */
+
+const sortContests = (contests, query) => {
+  let limit = Number(query.limit)
+  let active = Boolean(query.active) || false
+  let past = Boolean(query.past) || false
+  let upcoming = Boolean(query.upcoming) || false
+
+  if (!limit || isNaN(limit) || limit === null || limit === undefined) {
+    limit = 5
+  }
+  const today = new Date()
+  const currentTime = today.getTime()
+  let pastContests = []
+  let activeContests = []
+  let upcomingContests = []
+
+  let i = 0
+  while (i < contests.length && (active || past || upcoming)) {
```

Comment:
Is there any alternative for this.....it's going to iterate over all the contests which is highly not preferred....
instead, what i had in mind:

a `status` column in `contests` table which will auto update itself past `end_time` and b/w `start_time` and end_time`
status: INT NOT NULL
```
0: past
1: active
2: upcoming
```
using this...we can modify the query itself and wont need this `sortContests.js` at all

Reply:
We can better use the SQL time everywhere so that in query we can compare like start_time < NOW()  and similar.

Reply:
This seems like a complicated query.....ok let me try it in one single query